tests/integration/test_authd/test_api_agent_registration.py

```diff
@@ -0,0 +1,249 @@
+'''
+copyright: Copyright (C) 2015-2021, Wazuh Inc.
+
+           Created by Wazuh, Inc. <info@wazuh.com>.
+
+           This program is free software; you can redistribute it and/or modify it under the terms of GPLv2
+
+type: integration
+
+brief: These tests will check if the 'wazuh-authd' daemon correctly handles the enrollment requests
+       from the API.
+
+tier: 0
+
+modules:
+    - authd
+    - API
+
+components:
+    - manager
+
+daemons:
+    - wazuh-authd
+    - wazuh-api
+
+os_platform:
+    - linux
+
+os_version:
+    - Arch Linux
+    - Amazon Linux 2
+    - Amazon Linux 1
+    - CentOS 8
+    - CentOS 7
+    - CentOS 6
+    - Ubuntu Focal
+    - Ubuntu Bionic
+    - Ubuntu Xenial
+    - Ubuntu Trusty
+    - Debian Buster
+    - Debian Stretch
+    - Debian Jessie
+    - Debian Wheezy
+    - Red Hat 8
+    - Red Hat 7
+    - Red Hat 6
+
+references:
+    - https://documentation.wazuh.com/current/user-manual/registering/restful-api-registration.html
+
+tags:
+    - authd
+    - api
+'''
+import os
+import pytest
+import yaml
+import requests
+import re
+import ipaddress
+
+from wazuh_testing.tools.configuration import load_wazuh_configurations
+from wazuh_testing.tools.services import control_service
+from wazuh_testing.tools import API_LOG_FILE_PATH
+
+from wazuh_testing.api import get_api_details_dict
+from wazuh_testing.tools import CLIENT_KEYS_PATH
+from wazuh_testing.tools.file import truncate_file
+from wazuh_testing.tools.wazuh_manager import remove_all_agents
+
+
+# Marks
+pytestmark = [pytest.mark.linux, pytest.mark.tier(level=0)]
+
+# Configuration
+test_data_path = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'data')
+api_registration_requets_file = os.path.join(test_data_path, 'api_agent_registration_cases.yaml')
+daemons_handler_configuration = {'all_daemons': True}
+api_registration_requests = []
+with open(api_registration_requets_file) as tcases:
+    api_registration_requests = yaml.safe_load(tcases)
+
+parameters = [
+    {'IPV6': 'yes'},
+]
+metadata = [
+    {'ipv6': 'yes'},
+]
```

Comment:
We could test some cases with IPv6 disabled, or if it is going to be always enabled, we could just set it.

Reply:
Talking with @cborla , it seems that the behaviour is the same with ipv6 enabled or disabled. This configuration block should be removed.